packages/hydrogen/src/customer/auth.helpers.ts

```diff
@@ -0,0 +1,253 @@
+export const userAgent =
+  'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36';
+
+export interface HydrogenSession {
+  get: (key: string) => string | undefined;
+  set: (key: string, value: string) => void;
+  unset: (key: string) => void;
+  commit: () => Promise<string>;
+}
+
+export interface AccessTokenResponse {
+  access_token: string;
+  expires_in: number;
+  id_token: string;
+  refresh_token: string;
+  error?: string;
+  error_description?: string;
+}
+
+export async function refreshToken(
+  session: HydrogenSession,
+  customerAccountId: string,
+  customerAccountUrl: string,
+  origin: string,
+) {
+  const newBody = new URLSearchParams();
+
+  const refreshToken = session.get('refresh_token');
+
+  if (!refreshToken)
+    throw new BadRequest(
+      'Unauthorized',
+      'No refresh_token in the session. Make sure your session is configured correctly and passed to `createCustomerClient`.',
+    );
```

Comment:
It's super easy to get something small wrong in the oauth flow. In dev mode, I figured we can add these helper messages, potentially with links to a troubleshooting guide.